packages/repository/src/decorators/repository.decorator.ts

```diff
@@ -172,6 +172,13 @@ export function repository(
   };
 }
 
+export namespace repository {
+  // tslint:disable-next-line:no-any
+  export function getter<T = any>(bindingKey: BindingAddress<T>) {
```

Comment:
`@repository` support multiple flavors:
- `repository(repositoryName: string | Class<Repository<Model>>)`
- `repository(model: string | typeof Entity, dataSource: string | juggler.DataSource)`

I am ok to support only `bindingKey` flavor (i.e. `repository(repositoryName: string)`) in this pull request, we can add support for other flavors later.

However, I am wondering whether we should add some checks to throw a helpful error if this new decorator is injected in a way that's not supported? Unless the function signature is good enough to allow the TypeScript compiler to reject such programmer mistakes.

Few examples of incorrect usage that I am thinking of:
 - `@repository.getter(ProductRepository)` 
 - `@repository.getter('Product', dbDataSource)`
 - `@repository.getter(Product, 'db')`

Reply:
I believe all of the incorrect usage should be covered `BindingAddress`. It can be strengthened so that `T` can extend from `EntityCrudRepository` though.

Reply:
> I believe all of the incorrect usage should be covered BindingAddress.

Sounds good.